addons/show-bbcode/userscript.js

```diff
@@ -0,0 +1,58 @@
+function viewSource(post, msg) {
+  return function (event) {
+    event.preventDefault();
+    const body = post.querySelector(".postmsg");
+    if (event.target.getAttribute("data-state") === "post") {
```

Comment:
Shouldn't you check to see if it's already loading so you don't get multiple requests/dom changes happening at once?

Reply:
If it's already loading then the state would be `loading`, not `post`.

Reply:
Ah, ok, I understand now. I didn't see that earlier.